packages/ember-glimmer/lib/component.js

```diff
@@ -401,6 +876,68 @@ const Component = CoreView.extend(
      */
 
     /**
+      A component may contain a layout. A layout is a regular template but
+      supersedes the `template` property during rendering. It is the
+      responsibility of the layout template to retrieve the `template`
+      property from the component (or alternatively, call `Handlebars.helpers.yield`,
+      `{{yield}}`) to render it in the correct location.
+      This is useful for a component that has a shared wrapper, but which delegates
+      the rendering of the contents of the wrapper to the `template` property
+      on a subclass.
+      @property layout
+      @type Function
+      @public
+    */
+
+    /**
+      The name of the layout to lookup if no layout is provided.
+      By default `Ember.Component` will lookup a template with this name in
+      `Ember.TEMPLATES` (a shared global object).
+      @property layoutName
+      @type String
+      @default null
+      @private
+    */
+
+    /**
+      Returns a jQuery object for this component's element. If you pass in a selector
+      string, this method will return a jQuery object, using the current element
+      as its buffer.
+      For example, calling `component.$('li')` will return a jQuery object containing
+      all of the `li` elements inside the DOM element of this component.
+      @method $
+      @param {String} [selector] a jQuery-compatible selector string
+      @return {jQuery} the jQuery object for the DOM node
+      @public
+    */
+
+    /**
+      The HTML `id` of the component's element in the DOM. You can provide this
+      value yourself but it must be unique (just as in HTML):
+
+      ```handlebars
+        {{my-component elementId="a-really-cool-id"}}
+      ```
+      If not manually set a default value will be provided by the framework.
+      Once rendered an element's `elementId` is considered immutable and you
+      should never change it. If you need to compute a dynamic value for the
+      `elementId`, you should do this when the component or element is being
+      instantiated:
+
+      ```javascript
+        export default Ember.Component.extend({
+          setElementId: function() {
```

Comment:
Using `.on` event handlers is less desired nowadays. Something like the following would be more up to date:

```javascript
init() {
  this._super(...arguments);

  var index = this.get('index');
  this.set('elementId', `component-id${index}`);
}
```